graylog2-server/src/main/java/org/graylog2/rest/GenericErrorCsvWriter.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.rest;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.dataformat.csv.CsvGenerator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import org.graylog2.plugin.rest.ApiError;
+import org.graylog2.plugin.rest.GenericError;
+import org.graylog2.plugin.rest.ValidationApiError;
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+@Provider
+@Produces(MoreMediaTypes.TEXT_CSV)
+public class GenericErrorCsvWriter implements MessageBodyWriter<GenericError> {
+    private final CsvMapper mapper;
+
+    public GenericErrorCsvWriter() {
+        this.mapper = new CsvMapper();
```

Comment:
We could probably move that out of this class, make it a singleton, and also use it in `ScrollChunkWriter`.